app/src/main/java/com/amaze/filemanager/utils/DataUtils.java

```diff
@@ -291,7 +291,12 @@ public void setHistory(LinkedList<String> s) {
     }
 
     public void addHistoryFile(final String i) {
-        history.push(i);
+        String d=i.replaceFirst("/+[^/]*","");
```

Comment:
Why are you doing this?

Reply:
I never find a file in  / or / */  directory. 
also, opening from / directory list is very fast. 
I think it is a waste of screen and attention.

Reply:
I don't think removing valid folders from history is a good idea. I think it as a browser history, you don't remove link just because they are easily accessible...